frontend/packages/patternfly/src/components/notification-drawer/notification-entry.tsx

```diff
@@ -36,59 +36,76 @@ const NotificationIcon: React.FC<NotificationIconTypes> = ({ type }) => {
+      onClick={(e) => {
+        e.stopPropagation();
+        drawerToggle(e);
+      }}
+    >
+      {text}
+    </Link>
+  </div>
+);
+
 const NotificationEntry: React.FC<NotificationEntryProps> = ({
+  actionText,
+  actionURL,
   title,
   description,
   isRead = false,
   timestamp,
   targetURL,
   toggleNotificationDrawer,
   type,
-}) => {
-  const notificationAction = type === NotificationTypes.update && targetURL && (
-    <div className="pf-c-notification-drawer__header-action">
-      <Link to="">Install Update</Link>
+}) => (
+  <li
+    className={classNames(`pf-c-notification-drawer__list-item pf-m-hoverable pf-m-${type}`, {
+      'pf-m-read': isRead,
+    })}
+    tabIndex={0}
+    onClick={
+      targetURL
+        ? () => {
+            history.push(targetURL);
+            toggleNotificationDrawer();
+          }
+        : null
+    }
+  >
+    <div className="pf-c-notification-drawer__list-item-header">
+      <span className="pf-c-notification-drawer__list-item-header-icon">
+        <NotificationIcon type={type} />
+      </span>
+      <h4 className="pf-c-notification-drawer__list-item-header-title">
+        <span className="pf-screen-reader">{`${_.capitalize(type)} notification:`}</span>
+        {title}
+      </h4>
+      {actionText && actionURL && (
+        <NotificationAction
+          text={actionText}
+          url={actionURL}
+          drawerToggle={toggleNotificationDrawer}
+        />
+      )}
     </div>
-  );
-  return (
-    <li
-      className={classNames(`pf-c-notification-drawer__list-item pf-m-hoverable pf-m-${type}`, {
-        'pf-m-read': isRead,
-      })}
-      tabIndex={0}
-      onClick={
-        targetURL
-          ? () => {
-              history.push(targetURL);
-              toggleNotificationDrawer();
-            }
-          : null
-      }
-    >
-      <div className="pf-c-notification-drawer__list-item-header">
-        <span className="pf-c-notification-drawer__list-item-header-icon">
-          <NotificationIcon type={type} />
-        </span>
-        <h4 className="pf-c-notification-drawer__list-item-header-title">
-          <span className="pf-screen-reader">{`${_.capitalize(type)} notification:`}</span>
-          {title}
-        </h4>
-        {notificationAction}
-      </div>
-      <div className="pf-c-notification-drawer__list-item-description">{description}</div>
-      <div className="pf-c-notification-drawer__list-item-timestamp">
-        {timestamp && <Timestamp simple timestamp={timestamp} />}
-      </div>
-    </li>
-  );
-};
+    <div className="pf-c-notification-drawer__list-item-description">{description}</div>
+    <div className="pf-c-notification-drawer__list-item-timestamp">
+      {timestamp && <Timestamp simple timestamp={timestamp} />}
+    </div>
+  </li>
+);
 
 export type NotificationEntryProps = {
-  title: string;
+  actionText?: string;
+  actionURL?: string;
   description: string;
   isRead?: boolean;
   targetURL?: string;
```

Comment:
not new, but this isn't a URL, right?

```suggestion
  targetPath?: string;
```